src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -269,14 +268,44 @@ def __read_kv_from_app_service(cmd, appservice_account, prefix_to_add=""):
             cmd, resource_group_name=appservice_account["resource_group"], name=appservice_account["name"], slot=None)
         for item in settings:
             key = prefix_to_add + item['name']
-            value = item['value']
             tags = {'AppService:SlotSetting': str(item['slotSetting']).lower()} if item['slotSetting'] else {}
+            value = item['value']
+
+            # Value will look like one of the following if it is a KeyVault reference:
+            # @Microsoft.KeyVault(SecretUri=https://myvault.vault.azure.net/secrets/mysecret/ec96f02080254f109c51a1f14cdb1931)
+            # @Microsoft.KeyVault(VaultName=myvault;SecretName=mysecret;SecretVersion=ec96f02080254f109c51a1f14cdb1931)
+            if value and value.lower().startswith(KeyVaultConstants.APPSVC_KEYVAULT_PREFIX.lower()):
+                try:
+                    appsvc_value_dict = dict(x.split('=') for x in value[len(KeyVaultConstants.APPSVC_KEYVAULT_PREFIX) + 1: -1].split(';'))
+                    appsvc_value_dict_lower = {k.lower(): v for k, v in appsvc_value_dict.items()}
+                    secret_identifier = appsvc_value_dict_lower.get('secreturi')
+                    if not secret_identifier:
+                        # Construct secreturi
+                        vault_name = appsvc_value_dict_lower.get('vaultname')
+                        secret_name = appsvc_value_dict_lower.get('secretname')
+                        secret_version = appsvc_value_dict_lower.get('secretversion')
+                        secret_identifier = "https://{0}.vault.azure.net/secrets/{1}/{2}".format(vault_name, secret_name, secret_version)
+                    try:
+                        from azure.keyvault.key_vault_id import KeyVaultIdentifier
+                        # this throws an exception for invalid format of secret identifier
+                        KeyVaultIdentifier(uri=secret_identifier)
+                        appconfig_value = json.dumps({"uri": secret_identifier}, ensure_ascii=False, separators=(',', ':'))
+                        content_type = KeyVaultConstants.KEYVAULT_CONTENT_TYPE
+                        kv = KeyValue(key=key, value=appconfig_value, tags=tags, content_type=content_type)
```

Comment:
nit:
```
kv = KeyValue(key=key,
                        value=json.dumps({"uri": secret_identifier}, ensure_ascii=False, separators=(',', ':')),
                        tags=tags,
                        content_type=KeyVaultConstants.KEYVAULT_CONTENT_TYPE)
```

Reply:
done

---
In reply to: [364446718](https://github.com/Azure/azure-cli/pull/11773#discussion_r364446718) [](ancestors = 364446718)